src/main/java/ru/test/restservice/service/CompilerService.java

```diff
@@ -23,39 +24,48 @@
 
     private final ProjectRepository projectRepository;
 
-    /**
-     * Метод, обращающийся к командной строке для вызова latex компилятора
-     *
-     * @param file компилируемый файл
-     */
-    public CompilationResultDTO compileTexFile(FileItemDTO file, UUID projectId) {
-        Project project = projectRepository.findById(projectId);
+    public String executeTerminalCommand(String command, String pathToFile) {
         StringBuilder messageBuilder = new StringBuilder();
-        String message,
-                pathToFile = project.path + Paths.get(file.path).getParent();
         try {
             ProcessBuilder pBuilder = new ProcessBuilder();
             if (isWindows) {
-                // TODO: biber.exe
-                pBuilder.command("cmd.exe", "/c", "pdflatex --synctex=1 --interaction=nonstopmode \"" + file.name + "\"");
+                pBuilder.command("cmd.exe", "/c", command);
             } else {
-                pBuilder.command("sh", "-c", "pdflatex --synctex=1 --interaction=nonstopmode \"" + file.name + "\"");
+                pBuilder.command("sh", "-c", command);
             }
             pBuilder.directory(new File(pathToFile));
             Process proc = pBuilder.start();
             BufferedReader in = new BufferedReader(new InputStreamReader(proc.getInputStream()));
             String line;
             while ((line = in.readLine()) != null) {
-                System.out.println(line);
                 messageBuilder.append(line);
             }
             proc.waitFor();
             in.close();
-            message = messageBuilder.toString();
         } catch (Exception e) {
-            e.printStackTrace();
             throw new CompilationException(e.getMessage());
         }
+        return messageBuilder.toString();
+    }
+
+    /**
+     * Метод, обращающийся к командной строке для вызова latex компилятора
+     *
+     * @param file компилируемый файл
+     */
+    public CompilationResultDTO compileTexFile(FileItemDTO file, UUID projectId) throws IOException {
+        Project project = projectRepository.findById(projectId);
+        Files.deleteIfExists(Paths.get(project.path + file.path.replaceAll("\\.tex$", ".pdf")));
+        String message,
+                pdfLaTeX = "pdflatex --synctex=1 --interaction=nonstopmode \"" + file.name + "\"",
+                biber = "biber.exe \"" + file.name.replaceAll("\\.tex$", "") + "\"",
```

Comment:
рекомендую String.format:
```suggestion
                pdfLaTeX = String.format("pdflatex --synctex=1 --interaction=nonstopmode \"%s\"", file.name),
                biber = String.format("biber.exe \"%s\" ", file.name.replaceAll("\\.tex$", "")),
```
И скорее всего терминал хавает одинарные кавычки:
```suggestion
                pdfLaTeX = String.format("pdflatex --synctex=1 --interaction=nonstopmode '%s' ", file.name),
                biber = String.format("biber.exe '%s' ", file.name.replaceAll("\\.tex$", "")),
```

Reply:
С двойными норм робит

Reply:
и с одинарными, скорее всего
не хочешь проверять -- окей, но String.format имеет смысл использовать